pyart/retrieve/echo_class.py

```diff
@@ -123,3 +127,354 @@ def steiner_conv_strat(grid, dx=None, dy=None, intense=42.0,
+        radar object
+
+    Other Parameters
+    ----------------
+    mass_centers : matrix
+        The centroids for each variable and hydrometeor class in (nclasses,
+        nvariables)
+    weights : vector
+        The weight given to each variable.
+    refl_field, zdr_field, rhv_field, kdp_field, temp_field : str
+        Inputs. Field names within the radar object which represent the
+        horizonal reflectivity, the differential reflectivity, the copolar
+        correlation coefficient, the specific differential phase and the
+        temperature field. A value of None for any of these parameters will
+        use the default field name as defined in the Py-ART configuration
+        file.
+    hydro_field : str
+        Output. Field name which represents the hydrometeor class field.
+        A value of None will use the default field name as defined in the
+        Py-ART configuration file.
+
+    Returns
+    -------
+    hydro : dict
+        hydrometeor classification field
+
+    References
+    ----------
+    Besic, N., Figueras i Ventura, J., Grazioli, J., Gabella, M., Germann, U.,
+    and Berne, A.: Hydrometeor classification through statistical clustering
+    of polarimetric radar measurements: a semi-supervised approach,
+    Atmos. Meas. Tech., 9, 4425-4445, doi:10.5194/amt-9-4425-2016, 2016
+
+    """
+    nclasses = 9
+    nvariables = 5
+    lapse_rate = -6.5
+
+    # select the centroids as a function of frequency band
+    if mass_centers is None:
+        mass_centers = np.zeros((nclasses, nvariables))
+        # assign coefficients according to radar frequency
+        if 'frequency' in radar.instrument_parameters:
+            freq = radar.instrument_parameters['frequency']['data'][0]
+            # C band
+            if freq >= 4e9 and freq < 8e9:
+                freq_band = 'C'
+                # centroids derived for MeteoSwiss Albis radar
+                #                     Zh      ZDR     kdp   RhoHV   delta_Z
+                mass_centers[0, :] = [
+                    13.5829,  0.4063, 0.0497, 0.9868,  1330.3]  # DS
+                mass_centers[1, :] = [
+                    02.8453,  0.2457, 0.0000, 0.9798,  0653.8]  # CR
+                mass_centers[2, :] = [
+                    07.6597,  0.2180, 0.0019, 0.9799, -1426.5]  # LR
+                mass_centers[3, :] = [
+                    31.6815,  0.3926, 0.0828, 0.9978,  0535.3]  # GR
+                mass_centers[4, :] = [
+                    39.4703,  1.0734, 0.4919, 0.9876, -1036.3]  # RN
+                mass_centers[5, :] = [
+                    04.8267, -0.5690, 0.0000, 0.9691,  0869.8]  # VI
+                mass_centers[6, :] = [
+                    30.8613,  0.9819, 0.1998, 0.9845, -0066.1]  # WS
+                mass_centers[7, :] = [
+                    52.3969,  2.1094, 2.4675, 0.9730, -1550.2]  # MH
+                mass_centers[8, :] = [
+                    50.6186, -0.0649, 0.0946, 0.9904,  1179.9]  # IH/HDG
```

Comment:
Could these be made into a lookup table? Say a dictionary of frequency bands ('C', 'X', etc) with the mass_center arrays being the entries?  Might make the code more readable and easier to add additional centroids in the future.